smithy-swift-codegen/src/test/kotlin/HttpUrlPathMiddlewareTests.kt

```diff
@@ -0,0 +1,59 @@
+import io.kotest.matchers.string.shouldContainOnlyOnce
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.swift.codegen.model.AddOperationShapes
+
+class HttpUrlPathMiddlewareTests {
+    private var model = javaClass.getResource("http-binding-protocol-generator-test.smithy").asSmithy()
+    var newTestContext: TestContext
+    init {
+        newTestContext = newTestContext()
+        newTestContext.generator.generateSerializers(newTestContext.generationCtx)
+        newTestContext.generator.generateProtocolClient(newTestContext.generationCtx)
+        newTestContext.generator.generateDeserializers(newTestContext.generationCtx)
+        newTestContext.generator.generateCodableConformanceForNestedTypes(newTestContext.generationCtx)
+        newTestContext.generationCtx.delegator.flushWriters()
+    }
+    private fun newTestContext(): TestContext {
+        val settings = model.defaultSettings()
+        model = AddOperationShapes.execute(model, settings.getService(model), settings.moduleName)
+        return model.newTestContext()
+    }
```

Comment:
I don't think our unit tests should have instance variables.  Can we build this as a utility which avoids keeping state between execution of unit tests in this class?

Reply:
in this PR or a separate PR where we refactor like 80% of our unit tests that are setting things up like this?

Reply:
nvm I think I know what you want here, updating